src/psyclone/psyir/nodes/array_member.py

```diff
@@ -88,6 +91,128 @@ def create(member_name, indices):
             obj.addchild(child)
         return obj
 
+    def is_lower_bound(self, index):
+        '''Returns True if the specified array index contains a Range node
+        which has a starting value given by the 'LBOUND(access-expr,index)'
+        intrinsic where 'access-expr' is the full access to the current
+        ArrayMember and 'index' matches the specified array index. Otherwise
+        False is returned.
+
+        For example, if a Fortran array A%B was allocated with extent 10
+        then the starting value is 1 and LBOUND(A%B,1) would
+        return that value.
+
+        :param int index: the array index to check.
+
+        :returns: True if the array index is a range with its start \
+            value being LBOUND(array,index) and False otherwise.
+        :rtype: bool
+
+        '''
+        self._validate_index(index)
+
+        array_dimension = self.indices[index]
+        if not isinstance(array_dimension, Range):
+            return False
+
+        lower = array_dimension.children[0]
+        if not (isinstance(lower, BinaryOperation) and
+                lower.operator == BinaryOperation.Operator.LBOUND):
+            return False
+        if not isinstance(lower.children[0], Reference):
+            return False
+
+        # Check that the full structure access being passed to the LBOUND
+        # operator matches the one for this node.
+        return self._matching_access(lower.children[0])
+
+    def is_upper_bound(self, index):
+        '''Returns True if the specified array index contains a Range node
```

Comment:
Same as previous method: contains -> is, consider removing example (I am less sure about this), use semantic navigation when possible to make code clearer.

Reply:
Done (in array_mixin now).